bpf/bpf_sock.c

```diff
@@ -452,10 +454,17 @@ __sock4_health_fwd(struct bpf_sock_addr *ctx __maybe_unused)
 __section("cgroup/connect4")
 int cil_sock4_connect(struct bpf_sock_addr *ctx)
 {
+	int err;
+
 	if (sock_is_health_check(ctx))
 		return __sock4_health_fwd(ctx);
 
-	__sock4_xlate_fwd(ctx, ctx, false);
+	err = __sock4_xlate_fwd(ctx, ctx, false);
+	if (err == -EHOSTUNREACH || err == -ENOMEM) {
+		try_set_retval(err);
+		return SYS_REJECT;
```

Comment:
Couldn't we also do that for code that returns `SYS_REJECT` and existed before this commit? I see you did that for some BPF programs in a subsequent commit, but it seems it's still missing for `__sock{4,6}_health_fwd` and `cgroup/bind{4,6}`.